src/core/js/router.js

```diff
@@ -31,11 +31,21 @@ define([
 
         handleRoute: function() {
             var args = [].slice.call(arguments, 0, arguments.length);
-            if (arguments[arguments.length-1] === null) args.pop();
+
+            if (args.length) {
+                // Remove any null arguments, starting from the last.
+                for (var i = (args.length - 1); i >= 0; i--) {
```

Comment:
as mentioned you could probably use `_.without` here to exclude null arguments in just one line of code - but if you want to leave it as-is I'm fine with that